.github/workflows/cross-ci-build.yml

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright Debezium Authors
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+name: Cross CI Build
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout db2 repository
+        uses: actions/checkout@v2
+        with:
+          path: db2
+      - name: Check if pull request branch exists in debezium main repository
+        id: branch
+        env:
+          branch_name: ${{ github.head_ref }}
+        run: |
+            BRANCH=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/debezium/debezium/pulls | jq '.[] | {branch: .head.ref}')
+            if grep -q "$branch_name" <<< "$BRANCH"; then
+              echo "::set-output name=FOUND::true"
+            fi
+      - name: Checkout core repository @pull_request_branch
+        if: ${{ steps.branch.outputs.FOUND == 'true' }}
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.event.pull_request.user.login }}/debezium
+          ref: ${{ github.head_ref }}
+          path: core
+      - name: Checkout core repository @master
```

Comment:
```suggestion
      - name: Checkout core repository default branch
```

Reply:
>  Also do we know why the build failed?

Well, that was caused because the PR with branch `DBZ-3748` in the debezium main repository was last updated on July 14. But this PR branch is in sync with the current `main` branch of the db2 repository.

Reply:
>  Aside from that, do we have a document somewhere that outlines all the actions we use and their versions? I'm curious as we build more and more sophisticated action pipelines, how do we determine whether an action has been updated or even deprecated.

So far we don't have any document for GH actions. But I think we are going to need them as the number and complexity of actions are gradually increasing.